src/main/java/net/minecraftforge/common/extensions/IForgeRecipe.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2018.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.extensions;
+
+import java.util.Collection;
+
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraftforge.common.crafting.RecipeType;
+import net.minecraftforge.common.crafting.VanillaRecipeTypes;
+
+public interface IForgeRecipe {
+
+    /**
+     * Used when sorting this recipe into it's category during load, and in the default type matcher below.
+     * @return The type of this recipe.
+     */
+    default RecipeType<? extends IRecipe> getType()
+    {
+        return VanillaRecipeTypes.CRAFTING;
```

Comment:
In my opinion, we should not default implement this to ensure correctness. We can move this to recipe serializers and enforce recipe serializers to provide a recipe type.

Reply:
There's no good reason to move the type to the serializers.  The serializers are class typed, and the class has a recipe type.

Reply:
Hmm, I don't think you can access recipe types with classes?

Reply:
Again, I still believe we should enforce the type by adding an abstract method in the recipe serializer interface. This method is easily forgotten by implementations and will cause subtle issues.

Reply:
We shouldn't need to handhold people through the implementations.  Modders using this will likely be using direct IRecipe for crafting, and people who are using their own types should know to override this.